src/main/java/io/vertx/ext/dropwizard/impl/Matcher.java

```diff
@@ -15,14 +14,26 @@
 
```

Comment:
The implementation looks a bit complex to me. First we now have four fields to store the match definitions. Maybe that can be reduced. Besides we need to create matchers and perform matching twice in case we've defined matches with identifiers. Eventually, I would avoid Optional here in internal classes since this will create a lot of objects (this is going to be evaluated on every request).